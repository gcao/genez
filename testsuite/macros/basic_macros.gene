#!/usr/bin/env gene run

# Test: Basic macro definition and usage
(macro double [x]
  (+ %x %x))

(var a 5)
(print "double(5) = " (double a))

# Macro with print
(macro debug [msg value]
  (do
    (print "DEBUG: " %msg " = " %value)
    %value))

(var result (debug "calculation" (+ 10 20)))
(print "result = " result)