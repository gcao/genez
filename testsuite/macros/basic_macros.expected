double(5) =  10
DEBUG:  calculation  =  30
result =  30